x-pack/plugins/ml/public/explorer/explorer.js

```diff
@@ -52,48 +114,782 @@ function mapSwimlaneOptionsToEuiOptions(options) {
+              bounds,
+              getSwimlaneBucketInterval(this.props.jobs, swimlaneWidth),
+              false
+            );
+            const selectedJobIds = getSelectedJobIds(this.props.jobs);
+
+            // Load the overall bucket scores by time.
+            // Pass the interval in seconds as the swimlane relies on a fixed number of seconds between buckets
+            // which wouldn't be the case if e.g. '1M' was used.
+            // Pass 'true' when obtaining bucket bounds due to the way the overall_buckets endpoint works
+            // to ensure the search is inclusive of end time.
+            const overallBucketsBounds = getBoundsRoundedToInterval(
+              bounds,
+              getSwimlaneBucketInterval(this.props.jobs, swimlaneWidth),
+              true
+            );
+            mlResultsService.getOverallBucketScores(
+              selectedJobIds,
+              // Note there is an optimization for when top_n == 1.
+              // If top_n > 1, we should test what happens when the request takes long
+              // and refactor the loading calls, if necessary, to avoid delays in loading other components.
+              1,
+              overallBucketsBounds.min.valueOf(),
+              overallBucketsBounds.max.valueOf(),
+              getSwimlaneBucketInterval(this.props.jobs, swimlaneWidth).asSeconds() + 's'
+            ).then((resp) => {
+              this.skipCellClicks = false;
+              const overallSwimlaneData = processOverallResults(
+                resp.results,
+                searchBounds,
+                this.props.jobs,
+                swimlaneWidth
+              );
+
+              console.log('Explorer overall swimlane data set:', overallSwimlaneData);
+              const hasResults = (overallSwimlaneData.points && overallSwimlaneData.points.length > 0);
+              this.setState(
+                {
+                  hasResults,
+                  loading: false,
+                  overallSwimlaneData,
+                },
+                () => {
+                  if (this.state.hasResults) {
+                    // Trigger loading of the 'view by' swimlane -
+                    // only load once the overall swimlane so that we can match the time span.
+                    this.setViewBySwimlaneOptions(resolve);
+                  } else {
+                    resolve();
+                  }
+                }
+              );
+            });
+          }
+        );
+      });
+    }
+
+    setViewBySwimlaneOptions(resolve) {
+      // Obtain the list of 'View by' fields per job.
+      this.setState(
+        { swimlaneViewByFieldName: '' },
+        () => {
+          // Unique influencers for the selected job(s).
+          let viewByOptions = [];
+
+          const selectedJobIds = getSelectedJobIds(this.props.jobs);
+          const fieldsByJob = { '*': [] };
+          _.each(mlJobService.jobs, (job) => {
```

Comment:
Nit - could get rid of the use of lodash `each` here

Reply:
Done in 7ec1bfedaf.